snapm/manager/_manager.py

```
@@ -351,6 +364,196 @@ def _find_mount_point_for_devpath(devpath):
+            lockdir = join(os.environ["XDG_RUNTIME_DIR"], _SNAPSET_LOCK_DIR_UNPRIV)
+        else:
+            _log_warn(
+                "Could not find XDG_RUNTIME_DIR for snapset lock directory:"
+                " falling back to /run/user/{euid}"
+            )
+            if not exists(f"/run/user/{euid}"):
+                _log_warn(
+                    "Could not find /run/user/%s for snapset lock directory:"
+                    " falling back to /tmp"
+                )
+                if not exists("/tmp"):
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp"

(S108)